src/go/rpk/pkg/vcloud/yak/client.go

```diff
@@ -59,19 +54,39 @@ func (yc *YakClient) GetNamespaces() ([]*Namespace, error) {
 	if err != nil {
 		return nil, fmt.Errorf("error reading request body: %w", err)
 	}
-
-	// targetting HTTP codes 401, 403 which are used by yak
-	if resp.StatusCode > 400 && resp.StatusCode < 404 {
-		log.Debug(body)
-		return nil, ErrNotAuthorized
-	}
-	if resp.StatusCode != 200 {
-		return nil, fmt.Errorf("error retrieving resource, http code %d. %s", resp.StatusCode, body)
+	err = yc.handleErrResponseCodes(resp, body)
+	if err != nil {
+		return nil, err
 	}
+
 	var namespaces []*Namespace
 	err = json.Unmarshal(body, &namespaces)
 	if err != nil {
 		return nil, fmt.Errorf("error unmarshaling response. %w", err)
 	}
 	return namespaces, nil
 }
+
+func (yc *YakClient) get(token, url string) (*http.Response, error) {
+	log.Debugf("Calling yak api on url %s", url)
+	req, err := http.NewRequest(http.MethodGet, url, bytes.NewBuffer([]byte{}))
+	if err != nil {
+		return nil, fmt.Errorf("error creating new request. %w", err)
+	}
+	req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", token))
+	return http.DefaultClient.Do(req)
+}
+
+func (yc *YakClient) handleErrResponseCodes(
+	resp *http.Response, body []byte,
+) error {
+	// targetting HTTP codes 401, 403 which are used by yak
+	if resp.StatusCode > 400 && resp.StatusCode < 404 {
+		log.Debug(string(body))
+		return ErrNotAuthorized
+	}
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("error retrieving resource, http code %d. %s", resp.StatusCode, body)
+	}
+	return nil
+}
```

Comment:
These changes should go in the 1st commit, where you introduced the YakClient.

Reply:
Agree I was waiting until the second method to extract the code meaningfully and to be honest solving the merge conflicts makes me not want to do it. Do you feel strongly about doing that? :D